[5.0] docs(treelist, gantt): Add Column Menu documentation (#1711)
telerik/blazor-docs: components/gantt/gantt-tree/columns/menu.md

---
title: Column Menu
page_title: Gantt - Column Menu

> description: Use the Column Menu for the Gantt to show a menu that allows you to perform high-level column customization

Reviewer: ```suggestion
description: Use the Column Menu for the Gantt to show a menu that allows you to perform high-level column customization.
```

### Column Chooser

The Column Chooser in the Column Menu allows you to toggle the visibility of Gantt columns. By default, all columns are visible under the **Columns** section of the Column Menu. To expand the menu, click the **Columns** item.

The **Apply** button sets the column visibility according to the current checkbox values and closes the column menu. The **Reset** button reverts the checkbox values to their state when the column menu was opened. At this point, the user can start over, click **Apply**, or click outside the column menu to close it.

* To disable the column chooser, set the `ShowColumnChooser` parameter of the `<GanttColumnMenuSettings>` to `false`.
* To hide a column from the Column Chooser, set the `VisibleInColumnChooser` property of the column to `false`.

### Filtering

To control whether filtering is possible from the Column Menu, set the `FilterMode` parameter of the `GanttColumnMenuSettings` tag to a member of the `ColumnMenuFilterMode` enum:

* `None`—disables the filtering from the Column Menu. This is the recommended option if you use the [`FilterRow` mode]({%slug gantt-filter-row%}).
* `FilterMenu`—enables filtering from a filter menu.

> ### Groupable

Reviewer 2: Better use "Grouping" to make it consistent with the other headings. There is no need to match the feature name, because "Filtering" and "Frozen columns" don't either.

Author: The Groupable paragraph as a whole is a mistake on my part, as the Gantt does not have such a feature.

To allow grouping the Gantt from the Column Menu, set the `Groupable` parameter of the `GanttColumnMenuSettings` tag to `true`. This feature will group the component by the column you have opened the Column Menu from.

### Frozen Columns

To disable the locking and unlocking of a column from the Column Menu, set the `Lockable` parameter of the `GanttColumnMenuSettings` tag to `false`.

### Sorting

To remove the sorting option from the Column Menu, set the `Sortable` parameter of the `GanttColumnMenuSettings` tag to `false`.

> ### Reorderable

Reviewer 2: Reordering

        var random = new Random();

        for (int i = 1; i < 6; i++)
        {
            var newItem = new FlatModel()
                {
                    Id = LastId,
                    Title = "Employee  " + i.ToString(),
                    Start = new DateTime(2020, 12, 10 + i),
                    End = new DateTime(2020, 12, 11 + i),
                    PercentComplete = Math.Round(random.NextDouble(), 2)
                };

            Data.Add(newItem);
            var parentId = LastId;
            LastId++;

            for (int j = 0; j < 5; j++)
            {
                Data.Add(new FlatModel()
                    {
                        Id = LastId,
                        ParentId = parentId,
                        Title = "    Employee " + i + " : " + j.ToString(),
                        Start = new DateTime(2020, 12, 20 + j),
                        End = new DateTime(2020, 12, 21 + i + j),
                        PercentComplete = Math.Round(random.NextDouble(), 2)
                    });

                LastId++;
            }
        }

        base.OnInitialized();
    }
}
````

### Column Menu Features Example

> >caption Use the GanttColumnMenuSettings tag to control the common features of the Column Menu, use column parameters to affect its relationship with the column menu

Author: By "common" features, I meant the previously mentioned features of the Column Menu, but on second thought, "common" might not be needed and "features" works by itself too.

    {
        Data = new List<FlatModel>();
        var random = new Random();

        for (int i = 1; i < 6; i++)
        {
            var newItem = new FlatModel()
                {
                    Id = LastId,
                    Title = "Employee  " + i.ToString(),
                    Start = new DateTime(2020, 12, 10 + i),
                    End = new DateTime(2020, 12, 11 + i),
                    PercentComplete = Math.Round(random.NextDouble(), 2)
                };

            Data.Add(newItem);
            var parentId = LastId;
            LastId++;

            for (int j = 0; j < 5; j++)
            {
                Data.Add(new FlatModel()
                    {
                        Id = LastId,
                        ParentId = parentId,
                        Title = "    Employee " + i + " : " + j.ToString(),
                        Start = new DateTime(2020, 12, 20 + j),
                        End = new DateTime(2020, 12, 21 + i + j),
                        PercentComplete = Math.Round(random.NextDouble(), 2)
                    });

                LastId++;
            }
        }

        base.OnInitialized();
    }
}
````

## Notes

> * Applying settings to a Gantt column like `Lockable="false"` will take precedence over the common settings applied in the `<GanttColumnMenuSettings>` and disable the above-mentioned functionalitiy for the corresponding column.

Reviewer: ```suggestion
* Applying settings like `Lockable="false"` to a Gantt column takes precedence over the common settings applied in the `<GanttColumnMenuSettings>` and disables the above-mentioned functionality for the corresponding column.
```

> * Applying settings to a Gantt column like `Lockable="false"` will take precedence over the common settings applied in the `<GanttColumnMenuSettings>` and disable the above-mentioned functionalitiy for the corresponding column.

Reviewer: What is the "above-mentioned functionality"?

Author: "Above mentioned functionality" is used to describe Lockable in this case.

> * If you are using the [Column Chooser Template]({%slug gantt-templates-column-chooser%}) or you are grouping the columns into [sections](#sections), it is recommended to add the `Title` parameter to all Gantt Columns.

Reviewer: ```suggestion
* When using the [Column Chooser Template]({%slug gantt-templates-column-chooser%}) or grouping the columns into [sections](#sections), add the `Title` parameter to all Gantt Columns.
```
